src/rust/catnap/futures/push.rs

```diff
@@ -66,21 +63,37 @@ impl Future for PushFuture {
     /// Polls the target [PushFuture].
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
         let self_: &mut PushFuture = self.get_mut();
-        match socket::send(self_.fd, &self_.buf[..], socket::MsgFlags::empty()) {
+        match unsafe {
+            libc::sendto(
```

Comment:
Why sendto instead of send here?

Reply:
For two reasons:
1. To be able to set the `MSG_DONTWAIT` flag (see my reply from above)
2. To intentionally align up the logic of `PushFuture` and `PopFuture`.

For (2) I do have a change on my queue that unifies the logic of both co-routines.

Reply:
Okay.